resources/js/Components/Detail/CampaignDetail/CampaignDetailBox.tsx

```diff
@@ -13,59 +17,100 @@ const StyledTypography = styled(Typography)({
 }
 
-const CampaignDetailBox = () => (
-    // {campaign}: CampaignDetailBoxPropType
-    <Box sx={{p: 5, backgroundColor: 'primary'}}>
-        <Grid container columnSpacing={1} rowSpacing={6} xs={6}>
-            <Grid item xs={12}>
-                <StyledTypography>NAME</StyledTypography>
-                <Typography>
-                    {/* {campaign.title} */}
-                    Campaign 1
-                </Typography>
-            </Grid>
-            <Grid item xs={12}>
-                <StyledTypography>SESSIONS PLAYED</StyledTypography>
-                <Typography>
-                    {/* {campaign.sessions_played} */}
-                    25
-                </Typography>
-            </Grid>
-            <Grid item xs={12}>
-                <StyledTypography>CHARACTERS</StyledTypography>
-                <Typography>
-                    Character 1, Character 27, Character 31, Character 45
-                </Typography>
-            </Grid>
-            <Grid item>
-                <Stack spacing={3} direction='row'>
-                    <Button
-                        variant='contained'
-                        startIcon={<CreateIcon fontSize='small' />}>
-                        INVITE
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<CreateIcon fontSize='small' />}>
-                        UPDATE
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<HistoryEduIcon fontSize='small' />}>
-                        ENTRY
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<AutoStoriesIcon fontSize='small' />}>
-                        DM Entry
-                    </Button>
-                </Stack>
-            </Grid>
-        </Grid>
-    </Box>
-)
+const CampaignDetailBox = ({
+    campaign,
+    userCharacter,
+    setIsEditDrawerOpen,
+}: CampaignDetailBoxPropType) => {
+    const [openInviteSnackbar, setOpenInviteSnackbar] = useState(false)
+
+    const handleClick = () => {
+        navigator.clipboard.writeText(campaign.code)
+        setOpenInviteSnackbar(true)
+    }
+
+    const handleClose = (event: any, reason: any) => {
+        if (reason === 'clickaway') {
+            return
+        }
+        setOpenInviteSnackbar(false)
+    }
+
+    return (
+        <>
+            <Snackbar
+                open={openInviteSnackbar}
+                autoHideDuration={6000}
+                onClose={handleClose}
+                message='Invite code copied to clipboard.'
+            />
+            <Box sx={{p: 5, backgroundColor: 'primary'}}>
+                <Grid container columnSpacing={1} rowSpacing={6} xs={6}>
+                    <Grid item xs={12}>
+                        <StyledTypography>NAME</StyledTypography>
```

Comment:
translation

Reply:
the rest of the file as well, add the key value pairs in both the english and french translation json